code/datums/flock/flockprotection.dm

```diff
@@ -1,76 +1,92 @@
+/// Makes the flockmind 'protect' things by marking them as an enemy if a flockdrone spots them doing it.
 /datum/component/flock_protection
-	var/flockdrones_can_hit
-	var/report_hand_attack
-	var/report_obj_attack
+	/// Do we stop flockdrones from harming it?
+	var/flock_immune
+	/// Do we get mad if someone punches it?
+	var/report_unarmed
+	/// Do we get mad if someone hits it with something?
+	var/report_attack
+	/// Do we get mad if someone throws something at it?
+	var/report_thrown
+	/// Do we get mad if someone shoots it?
+	var/report_proj
 
-/datum/component/flock_protection/Initialize(flockdrones_can_hit, report_hand_attack, report_obj_attack)
-	src.flockdrones_can_hit = flockdrones_can_hit
-	src.report_hand_attack = report_hand_attack
-	src.report_obj_attack = report_obj_attack
+/datum/component/flock_protection/Initialize(flock_immune=TRUE, report_unarmed=TRUE, report_attack=TRUE, report_thrown=TRUE, report_proj=TRUE)
+	src.flock_immune = flock_immune
+	src.report_unarmed = report_unarmed
+	src.report_attack = report_attack
+	src.report_thrown = report_thrown
+	src.report_proj = report_proj
 
 /datum/component/flock_protection/RegisterWithParent()
 	RegisterSignal(parent, COMSIG_ATTACKHAND, .proc/handle_attackhand)
 	RegisterSignal(parent, COMSIG_ATTACKBY, .proc/handle_attackby)
+	RegisterSignal(parent, COMSIG_ATOM_HITBY_THROWN, .proc/handle_hitby_thrown)
+	RegisterSignal(parent, COMSIG_ATOM_HITBY_PROJ, .proc/handle_hitby)
 
+/// Protect against punches/kicks/etc.
 /datum/component/flock_protection/proc/handle_attackhand(source, mob/user as mob)
 	if (user.a_intent != INTENT_HARM)
 		return
 
-	if (istype(user, /mob/living/critter/flock/drone) && !flockdrones_can_hit)
-		if (istype(source, /obj/storage/closet/flock) || istype(source, /turf/simulated/floor/feather) || istype(source, /obj/machinery/door/feather))
-			return
+	if (isflock(user) && src.flock_immune)
```

Comment:
Means message will be given even though no damage can be done for the closet, turf, and door, when humans can use harm intent with no damage or report given. Not sure how much it matters.